### PR TITLE
fix: 3000 styles in the toolbar

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

A number of issues with 3000 styles in the toolbar (explained in the Changes)

## Changes

- We no longer rely on the `posthog-3000` class for styles anywhere in the app / toolbar. Removing this and renaming `use3000Body` to `useThemedBody` to future-ify it
- Replaced some instances where we were hard coding `text-white` to use the 3000 supported `text-default`
- Added `root` support to the tooltip so that it renders in the correct context
- Fixed z-indices in the toolbar so that they maintain same ordering as in the app
- Use `useState` in the FloatingContainerContext to ensure re-renders. In the case of the toolbar a re-render was never happening so the `FloatingPortal root={floatingContainer}` would always be `null` and hence default to the body. This meant that CSS variables were unset and not using those specified in the toolbar. In the case of our own site the CSS values were actually coming from whatever theme you had defined on the app

## How did you test this code?

|Before|After|
|----|----|
| Screenshot 2024-02-22 at 16 24 11 | Screenshot 2024-02-22 at 14 34 58 |

|Before|After|
|----|----|
| Screenshot 2024-02-22 at 16 23 04 | Screenshot 2024-02-22 at 16 28 53 |

### CODE REVIEW

File: frontend/__snapshots__/components-html-elements-display--editable-display-with-preselection--dark.png
Comment (pauldambra) [inline frontend/__snapshots__/components-html-elements-display--editable-display-with-preselection--dark.png:1]: interestingly the class highlighting isn't working here... e.g. LemonButton__content should be highlighted

we can follow-up if necessary though instead of blocking here - since this is already much better (and broken the same way in the before shot)

File: frontend/__snapshots__/components-html-elements-display--empty-display--dark.png
Comment (pauldambra) [inline frontend/__snapshots__/components-html-elements-display--empty-display--dark.png:1]: looks like we've lost text or background color here (and in the other variants of the same)

File: frontend/src/lib/hooks/useFloatingContainerContext.ts [hunk 1]
```diff
@@ -11,6 +11,13 @@ import { createContext, RefObject, useContext } from 'react'
...
-    return useContext(FloatingContainerContext)
+export const useFloatingContainer = (): HTMLElement | null | undefined => {
+    const ref = useContext(FloatingContainerContext)
+    const [el, setEl] = useState<HTMLElement | null | undefined>(undefined)
+
+    useEffect(() => {
+        setEl(ref?.current)
+    }, [ref])
+
+    return el
```
Comment (pauldambra) [inline frontend/src/lib/hooks/useFloatingContainerContext.ts:22]: ELI5 why switch to returning the el instead of the ref?

Comment (pauldambra) [reply frontend/src/lib/hooks/useFloatingContainerContext.ts:22]: Ah, cos we were always just using the el whenever we returned the ref 🙌

**Review by pauldambra — APPROVED** [review]
didn't run it but the screenshot changes give me confidence 🚀

File: frontend/src/lib/hooks/useFloatingContainerContext.ts (hunk 1 above)
Comment (daibhin) [reply frontend/src/lib/hooks/useFloatingContainerContext.ts:22]: Yeah, it is also because the component doesn't re-render from a change in `useContext`, we need the `useState` value to be updated
